File: Java Fundamentals/Methods/src/PasswordValidator.java
import java.util.Scanner;

public class PasswordValidator {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String inputPassword = scan.nextLine();

        checkPasswordValidity(inputPassword);
    }

    public static void checkPasswordValidity(String password){

        boolean isValid = true;

        int numOfChars = 0;
        boolean isLettersOrDigits = true;
        int numOfDigits = 0;

        for (int i = 0; i < password.length(); i++){
            char currentChar = password.charAt(i);
            numOfChars++;

            if((currentChar >= 48 && currentChar <= 57) ||
               (currentChar >= 65 && currentChar <= 90) ||
               (currentChar >= 97 && currentChar <= 122)){
                //do nothing
            }else{
                isLettersOrDigits = false;
            }

            if(currentChar >= 48 && currentChar <= 57){
                numOfDigits++;
            }
        }

        if(numOfChars < 6 || numOfChars > 10){
            System.out.println("Password must be between 6 and 10 characters");
            isValid = false;
        }

        if(!isLettersOrDigits){
            System.out.println("Password must consist only of letters and digits");
            isValid = false;
        }

        if(numOfDigits < 2){
            System.out.println("Password must have at least 2 digits");
            isValid = false;
        }

        if(isValid){
            System.out.println("Password is valid");
        }
    }
}

File: JS Front-End/Functions and Statements/PasswordValidator.js
function validatePassword(password){

    function isLengthValid(password){
        return (password.length >= 6 && password.length <= 10);
    }

    function areCharsValid(password){
        let currentChar;
        for(let i = 0; i < password.length; i++){
            currentChar = password.charAt(i);

            if(!currentChar.match(/[0-9A-Za-z]/i)){
                return false;
            }
        }

        return true;
    }

    function areTwoDigitsPresent(password){
        let digitCount = 0;
        for(let i = 0; i < password.length; i++){
            if(password.charAt(i).match(/[0-9]/)){
                digitCount++;
            }
        }

        return digitCount >= 2;
    }

    let isValid = true;

    if(!isLengthValid(password)){
        console.log("Password must be between 6 and 10 characters");
        isValid = false;
    }

    if(!areCharsValid(password)){
        console.log("Password must consist only of letters and digits");
        isValid = false;
    }

    if(!areTwoDigitsPresent(password)){
        console.log("Password must have at least 2 digits");
        isValid = false;
    }

    if(isValid){
        console.log("Password is valid");
    }
}

validatePassword('logIn');
console.log("---");
validatePassword('MyPass123');
console.log("---");
validatePassword('Pa$s$s');